servers/display_server.cpp

```diff
@@ -34,6 +34,14 @@
 #include "scene/resources/texture.h"
 #include "servers/display_server_headless.h"
 
+#if defined(VULKAN_ENABLED)
+#include "drivers/vulkan/rendering_context_driver_vulkan.h"
+#undef CursorShape
```

Comment:
It won't compile without the `undef`. I guess because this is code is hit before any of those other places. 

I have tested on MacOS and linux and its definitely needed there. But, based on the fact it is included in `input.h`, I suspect it is worthwhile to undef on all platforms. That being said, I'm happy to restrict the `undef` to MacOS and Linux and see if builds fail if you think it is worth restricting.

Reply:
I suspect since this line is outside of the declaration of DisplayServer the compiler has no trouble treating `DisplayServer::CursorShape` as a different thing from `CursorShape`. 

If that's right, then another option would be to always write `DisplayServer::CursorShape` instead of just `CursorShape` in the body of the class. But that feels much grosser to me than undefing a define that is polluting the global namespace